### PR TITLE
Add list import modal in MailRbl settings.

### PR DESCRIPTION
**Pull request by adudekyeti**
Add list import modal in MailRbl settings.

### CODE REVIEW

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	 * @param string $content [$content description]
+	 * @param string $source
+	 *
+	 * @return [type] [return description]
+	 */
+	public function saveToDb(string $source, int $type, string $content): array
+	{
+		$firstLine = true;
+		$explodedElements = explode($this->delimiter, $content);
+		$db = \App\Db::getInstance('webservice');
```
Comment (rskrzypczak) [inline modules/Settings/MailRbl/actions/UploadList.php]: ?

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		$report = [
+			'saved' => 0,
+			'duplicates' => 0,
+			'errors' => 0
+		];
+		$source = \App\TextParser::textTruncate($source, 10);
+		foreach ($explodedElements as $elementToSave) {
+			$clearIp = \App\Purifier::purifyByType(trim($elementToSave), 'AlnumExtended');
+			if (filter_var($clearIp, FILTER_VALIDATE_IP)) {
```
Comment (rskrzypczak) [inline modules/Settings/MailRbl/actions/UploadList.php]:
```suggestion
			if (\App\Validator::ip($clearIp)) {
```

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+							'status' => 0,
+							'type' => $type,
+							'source' => $source,
+							'request' => 0,
+						])->execute();
+						++$report['saved'];
+					} else {
+						++$report['duplicates'];
+					}
+				} catch (\Exception $e) {
```
Comment (rskrzypczak) [inline modules/Settings/MailRbl/actions/UploadList.php]: ?

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		$response->emit();
+	}
+
+	/**
+	 * [saveToDb description].
+	 *
+	 * @param string $content [$content description]
+	 * @param string $source
+	 *
+	 * @return [type] [return description]
```
Comment (rskrzypczak) [inline modules/Settings/MailRbl/actions/UploadList.php]: ?

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: layouts/basic/modules/Settings/MailRbl/UploadListModal.tpl [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+						</div>
+					</div>
+					<div class="form-group row">
+						<div class="col-sm-3">
+							<label class="col-form-label">
+								{\App\Language::translate('LBL_LIST_TYPE', $QUALIFIED_MODULE)}
+							</label>
+						</div>
+						<div class="col-sm-9 controls">
+							<select class="select2" name="type" data-validation-engine='validate[required]'>
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Settings/MailRbl/UploadListModal.tpl]:
```suggestion
							<select class="select2" name="type" data-validation-engine="validate[required]">
```

File: layouts/basic/modules/Settings/MailRbl/UploadListModal.tpl [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+						</div>
+					</div>
+					<div class="form-group row">
+						<div class="col-sm-3">
+							<label class="col-form-label">
+								{\App\Language::translate('LBL_SELECT_LIST', $QUALIFIED_MODULE)}
+							</label>
+						</div>
+						<div class="col-sm-9 controls">
+							<input type="file" name="imported_list" accept=".txt" data-validation-engine='validate[required]' id="imported_list" />
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Settings/MailRbl/UploadListModal.tpl]:
```suggestion
							<input type="file" name="imported_list" accept=".txt" data-validation-engine="validate[required]" id="imported_list" />
```

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+ * @license	  YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Dudek <a.dudek@yetiforce.com>
+ */
+
+/**
+ * Settings MailRbl Upload List action class.
+ */
+class Settings_MailRbl_UploadList_Action extends Settings_Vtiger_Basic_Action
+{
+	public $delimiter = "\n";
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/actions/UploadList.php:23]: doc

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	 * @param   int     $type
+	 * @param   string  $content
+	 *
+	 * @return  array
+	 */
+	public function saveToDb(string $source, int $type, string $content): array
+	{
+		$firstLine = true;
+		$explodedElements = explode($this->delimiter, $content);
+		$db = \App\Db::getInstance();
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/actions/UploadList.php]:
```suggestion
		$db = \App\Db::getInstance('admin');
```

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+		$report = [
+			'saved' => 0,
+			'duplicates' => 0,
+			'errors' => 0
+		];
+		$source = \App\TextParser::textTruncate($source, 10);
+		foreach ($explodedElements as $elementToSave) {
+			$clearIp = \App\Purifier::purifyByType(trim($elementToSave), 'AlnumExtended');
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/actions/UploadList.php]: AlnumExtended ?

File: modules/Settings/MailRbl/actions/UploadList.php [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+			if (\App\Validator::ip($clearIp)) {
+				$isExists = (new \App\Db\Query())->from('s_#__mail_rbl_list')->where(['ip' => $clearIp])->exists();
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/actions/UploadList.php]:
```suggestion
				$isExists = (new \App\Db\Query())->from('s_#__mail_rbl_list')->where(['ip' => $clearIp])->exists($db);
```

File: modules/Settings/MailRbl/views/UploadListModal.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * YetiForce upload list Modal.
+ *
+ * @package   View
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/views/UploadListModal.php]:
```suggestion
 * @package   Settings.View
```

File: modules/Settings/MailRbl/views/UploadListModal.php [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+	 */
+	public $showFooter = false;
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function preProcessAjax(App\Request $request)
+	{
+		$this->qualifiedModuleName = $request->getModule(false);
+		$this->pageTitle = \App\Language::translate('LBL_UPLOAD_LIST', $this->qualifiedModuleName);
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/MailRbl/views/UploadListModal.php:34]: public $pageTitle = 'LBL_UPLOAD_LIST';

File: public_html/layouts/basic/modules/Settings/MailRbl/resources/UploadListModal.js [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+		/**
+		 * Register button send form
+		 * @param {jQuery} form
+		 */
+		registerUploadButton: function (form) {
+			form.on('submit', function (e) {
+				e.preventDefault();
+				if (form.validationEngine('validate') === true) {
+					app.removeEmptyFilesInput(form[0]);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Settings/MailRbl/resources/UploadListModal.js]: ?

File: public_html/layouts/basic/modules/Settings/MailRbl/resources/UploadListModal.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+						contentType: false
+					}).done(function (data) {
+						progressIndicatorElement.progressIndicator({ mode: 'hide' });
+						if (true === data.result.success) {
+							notifyType = 'success';
+						} else {
+							notifyType = 'error';
+						}
+						app.showNotify({
+							title: app.vtranslate('JS_MESSAGE'),
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Settings/MailRbl/resources/UploadListModal.js]: ?

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

**Pull request merged**